Source/lighting.cpp

```diff
@@ -114,26 +114,16 @@ uint8_t GetLight(Point position)
 	return dLight[position.x][position.y];
 }
 
-void DoUnLight(int nXPos, int nYPos, int nRadius)
+void DoUnLight(Point position, uint8_t radius)

```

Comment:
The argument named "radius" is not actually a radius.
Consider changing the arguments to a single `WordTileRectangle area`

Reply:
The way I read it it is a radius, but for simplicity the implementation updates the tiles in a bounding square rather then a circle. Since it's the implementation that makes use of a Rectangle I would find it a little odd to make the function take that as the input, especially since we do not have a Rectangle at the call site and would need to construct one any way then.

Reply:
? are we reading the same code.

The original code has this:
```cpp
	int minX = nXPos - nRadius;
	int maxX = nXPos + nRadius;
	int minY = nYPos - nRadius;
	int maxY = nYPos + nRadius;
```
That very much to me looks like it's using the point as the center. Which also aligns with how DoLight() applies full brightness to the given point which visually appears in the center of the light.

We also call it a radius in the RectableOf constructor:
https://github.com/diasurgical/devilutionX/blob/3304e8a5429e5158b26c2c492e0dc080796df06e/Source/engine/rectangle.hpp#L29

> And the radius of the circle inscribed into that rectangle is actually radius/2, which makes that argument more like a diameter than a radius.

Are you confusing radius and diameter?
https://en.wikipedia.org/wiki/Diameter
https://en.wikipedia.org/wiki/Radius

Reply:
There is this other constructor, which takes the top-left corner position and size.

https://github.com/diasurgical/devilutionX/blob/3304e8a5429e5158b26c2c492e0dc080796df06e/Source/engine/rectangle.hpp#L15

However, I agree with AJenbo that this code appears to be calling the constructor that takes center position and radius. We should be avoiding ambiguity since the following constructor for `SizeOf` is marked `explicit`, right?

https://github.com/diasurgical/devilutionX/blob/3304e8a5429e5158b26c2c492e0dc080796df06e/Source/engine/size.hpp#L22

Reply:
Ah, right, I confused it with the other constructor, sorry!